### PR TITLE
Add deprecations in CHANGELOG.md

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,7 +39,9 @@
 
   * [GenEvent] Hard deprecate `GenEvent` and provide alternatives in its docs
   * [Kernel] Using `()` to mean `nil` is deprecated
+  * [Kernel] `Atom.to_char_list/1`, `Float.to_char_list/1`, `Integer.to_char_list/1`, `Kernel.to_char_list/1`, `String.to_char_list/1` have been deprecated in favor of their `to_charlist/1` version. This aligns with the naming conventions in both Erlang and Elixir
   * [Module] Using Erlang parse transforms via `@compile {:parse_transform, _}` is deprecated
+  * [Typespec] `char_list/0` is deprecated in favor of `charlist/0`
```
Comment (eksperimental) [inline CHANGELOG.md]: Wondering whether we should refer to it as `t:char_list/0` or `char_list/0`?

Comment (josevalim) [inline CHANGELOG.md]: Let's be explicit: The type `char_list/0` is deprecated...

Comment (josevalim) [inline CHANGELOG.md]: Ping. :)

Comment (eksperimental) [inline CHANGELOG.md]: I will amend this and other changes once I am done with Deprecations, as there are changes that were left out in this changelog, if you think that's OK. if not I can change this and then submit a new PR

Comment (josevalim) [inline CHANGELOG.md]: Please take your time. I was just wondering if this fell through the cracks or not. :) Thank you!

**Comment by eksperimental** [comment]
Updated and ready to be merged

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**